File: paper_artifacts/experiments/llr9/artifacts/sources/kimi27sglang-fortran-skills/tsvc_2_s232__submitted.f90
subroutine tsvc_2_s232_fp64(aa, bb, LEN_2D, workspace, workspace_size) bind(C)
  use iso_c_binding
  implicit none
  integer(c_int64_t), value, intent(in) :: LEN_2D, workspace_size
  real(c_double), intent(inout) :: aa(LEN_2D, LEN_2D)
  real(c_double), intent(in) :: bb(LEN_2D, LEN_2D)
  type(c_ptr), value, intent(in) :: workspace
  integer(c_int64_t) :: j, i

  if (LEN_2D < 128) then
    do j = 2, LEN_2D
      do i = 2, j
        aa(i, j) = aa(i - 1, j) * aa(i - 1, j) + bb(i, j)
      end do
    end do
    return
  end if

  !$omp parallel do schedule(guided) private(i) proc_bind(close)
  do j = 2, LEN_2D
    do i = 2, j
      aa(i, j) = aa(i - 1, j) * aa(i - 1, j) + bb(i, j)
 end do
  end do
  !$omp end parallel do
end subroutine tsvc_2_s232_fp64

File: paper_artifacts/experiments/llr8/artifacts/sources/kimi27sglang-c-skills/tsvc_2_s232__submitted.c
#include <stdint.h>
#include <omp.h>

void tsvc_2_s232_fp64(double *restrict aa, const double *restrict bb, const int64_t LEN_2D) {
    #pragma omp parallel for schedule(nonmonotonic:guided)
    for (int64_t j = 1; j < LEN_2D; ++j) {
        double *restrict arow = aa + j * LEN_2D;
        const double *restrict brow = bb + j * LEN_2D;
        double prev = arow[0];
        for (int64_t i = 1; i <= j; ++i) {
            prev = prev * prev + brow[i];
            arow[i] = prev;
        }
    }
}
